modules/030-cloud-provider-azure/hooks/azure_cluster_configuration.go

```diff
@@ -0,0 +1,165 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"fmt"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/config"
+)
+
+type ProviderClusterConfigurationValues struct {
+	Provider     Provider `json:"provider"`
+	SSHPublicKey string   `json:"sshPublicKey"`
+}
+
+type Provider struct {
+	SubscriptionID string `json:"subscriptionId"`
+	ClientID       string `json:"clientId"`
+	ClientSecret   string `json:"clientSecret"`
+	TenantID       string `json:"tenantId"`
+	Location       string `json:"location"`
+}
+
+type ProviderDiscoveryDataValues struct {
+	Instances         Instances `json:"instances"`
+	Zones             []string  `json:"zones"`
+	ResourceGroupName string    `json:"resourceGroupName"`
+	VnetName          string    `json:"vnetName"`
+	SubnetName        string    `json:"subnetName"`
+}
+
+type Instances struct {
+	Urn            string            `json:"urn"`
+	DiskType       string            `json:"diskType"`
+	AdditionalTags map[string]string `json:"additionalTags"`
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 20},
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "provider_cluster_configuration",
+			ApiVersion: "v1",
+			Kind:       "Secret",
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"kube-system"},
+				},
+			},
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"d8-provider-cluster-configuration"},
+			},
+			FilterFunc: applyProviderClusterConfigurationSecretFilter,
+		},
+	},
+}, clusterConfiguration)
+
+func applyProviderClusterConfigurationSecretFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var secret = &v1.Secret{}
+	err := sdk.FromUnstructured(obj, secret)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert kubernetes object: %v", err)
+	}
+
+	return secret, nil
+}
+
+func clusterConfiguration(input *go_hook.HookInput) error {
+	secret := input.Snapshots["provider_cluster_configuration"][0].(*v1.Secret)
+
+	clusterConfigurationYAML := secret.Data["cloud-provider-cluster-configuration.yaml"]
```

Comment:
you can use helper hook https://github.com/deckhouse/deckhouse/blob/main/go_lib/hooks/cluster_configuration/hook.go

Reply:
i see that bash hook is simple. It copy provider cluster configuration and discovery cloud data.
And i do not see usage structures in another hooks.
I think we simplify this hook like in [yandex](https://github.com/deckhouse/deckhouse/blob/main/modules/030-cloud-provider-yandex/hooks/yandex_cluster_configuration.go)